### DISCUSSION TITLE
Get the last commit message and store it to an env variable in actions/checkout@v2

### DISCUSSION THREAD

**Discussion by sblantipodi** [community/community#25376] [Actions]
<p>As title.</p>
<p>How can I get the last commit message and store it to an env variable in actions/checkout@v2</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/sblantipodi">@sblantipodi</a> ,</p>
<p>The last commit message can be got from Github context :  <strong>${{ github.event.head_commit.message }}</strong> , and set it as env as below:</p>
<pre><code>- name: checkout
        uses: actions/checkout@v2
        env:
          message: ${{ github.event.head_commit.message }}
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181090682-0cf8b012-7bd7-440c-844a-b706656eb3dd.png" alt="commitmessage.png" data-base62-sha1="ieDofDkB6k9uyq97G9zLjop9RUO" width="353" height="230"></p>
<p>BTW, actions/checkout doesn’t need env setting since not related code are defined, the env won’t be invoked for the action, unless you rewrite the code to invoke the env variable.</p>
<p>Thanks.</p>

**Comment by sblantipodi** [comment]
<p>thanks for the answer but this code prints an empty message.</p>
<pre><code>- uses: actions/checkout@v2
      env:
        commitmsg: ${{ github.event.head_commit.message }}
    - name: get commit message
      run: |
          echo Commit MSG = $commitmsg
</code></pre>
<p>any help is much appreciated. :slight_smile: thanks.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/sblantipodi">@sblantipodi</a> ,</p>
<p>Thanks for your quick reply! This is a step level env which only works in checkout step.</p>
<p>If you’d like to use the env variable in the other steps, please set it as job level env, code sample as below:</p>
<pre><code>jobs:
  job1:
    env:
      commitmsg: ${{ github.event.head_commit.message }}
    runs-on: ubuntu-latest
    steps:
      - name: checkout
        uses: actions/checkout@v2
      - name: get commit message
        run: |
            echo Commit MSG = ${{ env.commitmsg }}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/369ff447a27bd1eabbbc0ea498b95f8e597c23eb.png" data-download-href="/uploads/short-url/7NetZtOwXOgHTvri2KbNJBod1fd.png?dl=1" title="commit message.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181090686-9c3decb4-7ce6-42f1-a434-6cf73c8f524a.png" alt="commit message.png" data-base62-sha1="7NetZtOwXOgHTvri2KbNJBod1fd" width="390" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/369ff447a27bd1eabbbc0ea498b95f8e597c23eb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">commit message.png</span><span class="informations">432×553 19.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Or you can use ‘<a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions" rel="nofollow noopener">set-env</a>’ to set/update the env variables. Code sample as below:</p>
<pre><code>- name: set the head commit message as env
        run: |
          echo '::set-env name=commitmsg::${{ github.event.head_commit.message }}'
      - name: checkout
        uses: actions/checkout@v2
      - name: get commit message
        run: |
            echo Commit MSG = $commitmsg
</code></pre>
<p>Thanks.</p>

**Comment by zanona** [comment]
<p>Hey, <a class="mention" href="/weide-zhou">@weide-zhou</a><br>
Just wondering if <code> github.event.head_commit.message</code> is documented anywhere?<br>
I looked it up here: <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows" rel="nofollow noopener">https://help.github.com/en/actions/reference/events-that-trigger-workflows</a> and here <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context</a> without luck :slight_smile:</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/zanona">@zanona</a>,</p>
<p>The value comes from github context, please check the doc here: <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context</a></p>
<p>You can export the github context with below code for your workflow:</p>
<pre><code class="lang-auto">      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
</code></pre>
<p>It will output github context as json format, then you can follow the structure to get the content you required.</p>
<p>Thanks.</p>

**Comment by ricardo-dematos** [comment]
Please notice that `${{ github.event.head_commit.message }}` is only available for `push` events.

**Reply by muzimuzhi** [reply]
And here's its doc, contents of the `push` webhook payload: https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#push.
